### DISCUSSION TITLE
Anyway to have a priority?

### DISCUSSION THREAD

**Discussion by lemonlambda** [goffrie/plex#51] [Q&A]
Like this for example
```rs
plex::parser! {
    fn parse_(Token, Span);

    // combine two spans
    (a, b) {
        Span {
            lo: a.lo,
            hi: b.hi,
        }
    }

    program: Program {
        statements[s] => Program { stmts: s }
    }

    statements: Vec<Expr> {
        => vec![],
        statements[mut st] outer[e] SemiColon => {
            st.push(e);
            st
        }
    }

    outer: Expr {
        #[priority]
        Token SemiColon => ...,
        inner[a] => a,
    }
    
    inner: Expr {
        Token => ....,
    }
}
```

**Comment by lemonlambda** [comment]
@goffrie

**Comment by maxrdz** [comment]
Requesting the same feature here, as I'm dealing with some issues in my grammar which could be solved with priority.

**Comment by goffrie** [comment]
Yes, there's the `#[overriding]` attribute that gives a rule precedence over other rules. There isn't any syntax for multiple priority levels but it is supported in the underlying LALR(1) library so it should be simple enough to add.

**Reply by maxrdz** [reply]
By 'it should be simple to add' are you referring to implementing it on the crate, or simple to add on the user's end by using the LALR crate directly? I see there is a #[overriding] attribute (even though I can't find docs on it) but it doesn't seem to apply any sort of precedence over other rules.

**Comment by goffrie** [comment]
I mean simple to add to this crate.
